crowbar_framework/app/models/service_object.rb

```diff
@@ -1168,6 +1168,9 @@ def apply_role(role, inst, in_queue)
     # of names of Chef roles.
     run_order = []
 
+    # get databag to remember potential removal of a role
+    databag = Chef::DataBag.load "crowbar/bc-template-#{@bc_name}"
```

Comment:
I asked this a few times already but I got no answer...
Thanks for the hint, I will use the proposal then :)